nulib/meadow
# Set collections.featured to not nullable with default = false

# Summary 

Set collections.featured to not nullable with default = false

# Specific Changes in this PR

- Add migration to modify `featured` column of `collections` table:
  - `up` updates `NULL` values to `false` and makes the column not nullable
  - `down` makes the column nullable but does not change any data

# Version bump required by the PR

See [Semantic Versioning 2.0.0](https://semver.org/) for help discerning which is required.


# Steps to Test

- Check structure and content of `collections` table in TablePlus
- `mix ecto.migrate`
- Check again and make sure the `featured` field is no longer nullable
- `mix ecto.rollback`
- Check again and make sure the `featured` field is nullable again

It might be necessary to close the table view/structure windows in TablePlus between migrations to prevent a locking error.

# :rocket: Deployment Notes

**Note** - __if you check any of these boxes go to the [(always open) `main` <- `staging` PR](https://github.com/nulib/meadow/pulls) and add detailed notes and instructions to help out others who may end up deploying your changes to production__

- Backward compatible API changes
- Backwards-incompatible API changes


# Tested/Verified
